src/Microsoft.AspNetCore.Mvc.Core/Builder/MvcAreaRouteBuilderExtensions.cs

```diff
@@ -128,10 +128,10 @@ public static class MvcAreaRouteBuilderExtensions
             }
 
             var defaultsDictionary = new RouteValueDictionary(defaults);
-            defaultsDictionary["area"] = areaName;
+            defaultsDictionary["area"] = defaultsDictionary.ContainsKey("area") ? (defaultsDictionary["area"]?? areaName) : areaName;
```

Comment:
I think this could be simplified even more:

```c#
defaultsDictionary["area"] = defaultsDictionary["area"] ?? areaName;
```

I'm quite certain that the indexer of the `RouteValueDictionary` doesn't throw if the key isn't present - it just returns `null`. Also note the formatting (spaces around all the binary operators).